### PR TITLE
Narrow API

### PR DESCRIPTION
**Pull request by phansys**
## Subject

Narrow API by using explicit type declarations.

I am targeting this branch, because these changes could break BC, although these cases aren't valid.

## Changelog

```markdown
### Added
- Added explicit return declarations in methods were they are missing and the return type was declared with docblocks.
```

## To do
    
- [ ] Check missing occurrences;
- [ ] Try to remove before the methods that should not be present on 4.0 (maybe another PR for that will be the better choice);
- [ ] Check if is possible to perform similar update for method arguments;
- [ ] Validate if getters can throw exceptions (like [`FormInterface::get()`](https://github.com/symfony/symfony/blob/288bc24d0362fe87c4eb5c67f89d08f9b93e5807/src/Symfony/Component/Form/Form.php#L932-L939)) instead of returning `null` for nonexistent values (by instance, `DatagridInterface::getFilter($name)`).

### CODE REVIEW

**Comment by core23** [comment]
Do you use any tools for this? If not, maybe you should work on a [rector](https://github.com/RectorPHP/Rector) level which can automate this steps in (our) other projects too

**Comment by phansys** [comment]
**phpdoc_to_return_type** and **no_superfluous_phpdoc_tags** rules from [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer#usage) were applied in this PR. The remaining changes were performed manually.
I'll try to check rector too, thank you :+1:

File: src/Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -1322,7 +1318,7 @@ public function getRootCode()
      *
      * @return AbstractAdmin the root admin class
      */
-    public function getRoot()
+    public function getRoot(): self
```
Comment (core23) [inline src/Admin/AbstractAdmin.php]: AdminInterface?

File: src/Admin/AbstractAdmin.php [hunk 2]
```diff
@@ -1614,19 +1597,19 @@ public function removeShowFieldDescription($name): void
...
-    public function getListFieldDescriptions()
+    public function getListFieldDescriptions(): array
     {
         $this->buildList();
 
         return $this->listFieldDescriptions;
     }
 
-    public function getListFieldDescription($name)
+    public function getListFieldDescription($name): FieldDescriptionInterface
```
Comment (core23) [inline src/Admin/AbstractAdmin.php]:
```suggestion
    public function getListFieldDescription(string $name): FieldDescriptionInterface
```

File: src/Admin/AbstractAdmin.php [hunk 3]
```diff
@@ -1614,19 +1597,19 @@ public function removeShowFieldDescription($name): void
...
     {
         return $this->hasListFieldDescription($name) ? $this->listFieldDescriptions[$name] : null;
     }
 
-    public function hasListFieldDescription($name)
+    public function hasListFieldDescription($name): bool
```
Comment (core23) [inline src/Admin/AbstractAdmin.php]:
```suggestion
    public function hasListFieldDescription(string $name): bool
```

File: src/Admin/AbstractAdmin.php [hunk 4]
```diff
@@ -2582,10 +2535,8 @@ protected function configure(): void
 
     /**
      * Returns a list of default filters.
-     *
-     * @return array
      */
-    final protected function getDefaultFilterValues()
+    final protected function getDefaultFilterValues(): array
```
Comment (core23) [inline src/Admin/AbstractAdmin.php]: This can be done on the stable branch

File: src/Admin/AbstractAdmin.php (hunk 2 above)
Comment (phansys) [reply src/Admin/AbstractAdmin.php]: It's part of the **TODO** list:
>Check if is possible to perform similar update for method arguments;

By the moment, this PR is only focused on return declarations. I'd like to do the same with method arguments in an automated way.

**Pull request closed without merging**